runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -345,63 +346,97 @@ MM_IndexableObjectAllocationModel::getSparseAddressAndDecommitLeaves(MM_Environm
 	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(envBase);
 	GC_ArrayObjectModel *indexableObjectModel = &extensions->indexableObjectModel;
 	const uintptr_t regionSize = extensions->heapRegionManager->getRegionSize();
-
 	uintptr_t byteAmount = 0;
 
 	/* Determine how many bytes to allocate outside of the spine (in arraylet leaves). */
 	Assert_MM_true(_allocateDescription.getBytesRequested() >= _allocateDescription.getContiguousBytes());
 	uintptr_t bytesRemaining = _allocateDescription.getBytesRequested() - _allocateDescription.getContiguousBytes();
+#define REGION_RESERVE_THRESHOLD 64
```

Comment:
comment: reservation of all regions in the main heap is does before we allocate them in Offheap. Since the reservation can fail we don't want set AC pointers directly in Offheap, but only in the temp array, and after flush it in Offheap

Reply:
reminder to update the comment